src/include/rbd_types.h

```diff
@@ -106,4 +107,9 @@ struct rbd_obj_header_ondisk {
 	struct rbd_obj_snap_ondisk snaps[0];
 } __attribute__((packed));
 
+enum group_image_link_state {
+	LINK_NORMAL,
```

Comment:
... and do you need a DETACHING state in order to protect against a crash while removing?